pkg/kn/commands/trigger/update_flags.go

```diff
@@ -51,6 +51,12 @@ func (f *TriggerUpdateFlags) GetUpdateFilters() (map[string]string, []string, er
 //Add is to set parameters
 func (f *TriggerUpdateFlags) Add(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&f.Broker, "broker", "default", "Name of the Broker which the trigger associates with.")
+	// The Sugar controller was integrated into main Eventing controller. With that the default behavior was changed as well.
+	// Users need to configure 'Automatic Broker Creation' per linked docs.
+	// Deprecated in 1.4, remove in 1.6.
 	cmd.Flags().BoolVar(&f.InjectBroker, "inject-broker", false, "Create new broker with name default through common annotation")
+	cmd.Flags().MarkDeprecated("inject-broker", "effective since 1.4 and will be removed in 1.6 release. \n"+
+		"Please refer to 'Automatic Broker Creation' section for configuration options, "+
+		"https://knative.dev/docs/eventing/sugar/#automatic-broker-creation.")
```

Comment:
Maybe add the recommendation to create a Broker directly via `kn broker create`, because that is what a user can do. I think we shouldn't really think about automatic broker creation on this level.

So maybe this a description:

```
Automatic broker creation is deprecated and might not work anymore for Knative >= 1.4 (depending on server-side configuration). Please manage brokers directly with the `kn broker` commands.
```

I wouldn't mention that this is deprecated since client 1.4, I think its good enough to know that it's deprecated for the currently used version ;-). Also wouldn't put in links in the help message if not absolutely needed as we can't be sure that this link will stay stable over time.

Reply:
Sure, I was thinking about how stable those links might be. 

Does it look better? (the prefix is part of cobra `MarkDeprecated` function)

```console
Flag --inject-broker has been deprecated, automatic broker creation might not work anymore for Knative >= 1.4 (depending on server-side configuration). 
Please manage brokers directly with the `kn broker` commands.
```

Reply:
looks good, thanks !

/approve
/lgtm